### PR TITLE
[GEOS-9779][GEOS-9780] Upgrade OSHI from 3.13.3 to 5.3.3

### PR DESCRIPTION
**Pull request by mprins**
- [x] Upgrade OSHI from 3.13.3 to 5.3.3
- [x] mark all OSHI class members in OSHISystemInfoCollector `transient` are they are not serializable
- [x] fix API changes and deprecations

Resolves:
- [GEOS-9779](https://osgeo-org.atlassian.net/browse/GEOS-9779) Upgrade OSHI libraries to 5.3.3
- [GEOS-9780](https://osgeo-org.atlassian.net/browse/GEOS-9780) status page shows (read-only) loopback partitions

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

**Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or inapplicable.**

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/master/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)
- [x] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.

The following are required only for core and extension modules (they are welcomed, but not required, for community modules):
- [x] There is a ticket in Jira describing the issue/improvement/feature (a notable exemptions is, changes not visible to end users)
- [x] PR for bug fixes and small new features are presented as a single commit
- [x] Commit message must be in the form "[GEOS-XYZW] Title of the Jira ticket" (export to XML in Jira generates the message in this exact form)
- [x] The pull request contains changes related to a single objective. If multiple focuses cannot be avoided, each one is in its own commit and has a separate ticket describing it.
- [ ] New unit tests have been added covering the changes
- [x] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [x] This PR passes the [QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html) (QA checks results will be reported by travis-ci after opening this PR)
- [ ] Commits changing the UI, existing user workflows, or adding new functionality, need to include documentation updates (screenshots, text)
- [ ] Commits changing the REST API, or any configuration object, should check if the REST API docs (Swagger YAML files and classic documentation) need to be updated.

### CODE REVIEW

**Comment by aaime** [comment]
@nmco believe this is up your alley? Or is it @fernandor777 's ?

**Comment by aaime** [comment]
BTW, werent' there licensing changes in OSHI making it compatible with GPL? If so, we could remove the exemption here: https://github.com/geoserver/geoserver/blob/master/LICENSE.txt#L14

**Comment by mprins** [comment]
not sure about any recent changes in licensing, currently OSHI has a MIT "license" (actually that is a copyright + non-endorsement only): https://github.com/oshi/oshi/blob/master/LICENSE.txt (and had so for 3.x as well, old versions (1.3.. / 2015) seem to have has EPL 1 but those predate any use in GeoServer)

**Comment by mprins** [comment]
seeing an error in the geoserver log that I need to address:

```
2020-10-26 16:17:04,067 ERROR [java.JavaSerializer] - Error serializing object class org.geoserver.web.admin.StatusPage [object=[Page class = org.geoserver.web.admin.StatusPage, id = 11, render count = 1]]
org.apache.wicket.core.util.objects.checker.CheckingObjectOutputStream$ObjectCheckException: The object type is not Serializable!
A problem occurred while checking object with type: oshi.software.os.linux.LinuxFileSystem
Field hierarchy is:
  11 [class=org.geoserver.web.admin.StatusPage, path=11]
    private java.lang.Object org.apache.wicket.MarkupContainer.children [class=java.util.ArrayList]
      java.lang.Object org.apache.wicket.Component.data[write:90][write:92] [class=org.apache.wicket.extensions.ajax.markup.html.tabs.AjaxTabbedPanel, path=11:tabs]
        private java.lang.Object org.apache.wicket.MarkupContainer.children [class=java.util.ArrayList]
          private org.apache.wicket.markup.html.link.PopupSettings org.apache.wicket.markup.html.link.Link.popupSettings[write:2][write:5] [class=org.geoserver.web.system.status.SystemStatusMonitorPanel, path=11:tabs:panel]
            private java.lang.Object org.apache.wicket.MarkupContainer.children [class=org.geoserver.web.system.status.RefreshedPanel, path=11:tabs:panel:refreshed-values]
              java.lang.Object org.apache.wicket.Component.data [class=[Ljava.lang.Object;]
                java.lang.Object org.apache.wicket.Component.data[0] [class=org.apache.wicket.ajax.AjaxSelfUpdatingTimerBehavior]
                  final org.geoserver.system.status.SystemInfoCollector org.geoserver.web.system.status.RefreshedPanel$2.val$systemInfoCollector [class=org.geoserver.system.status.OSHISystemInfoCollector]
                    private oshi.software.os.FileSystem org.geoserver.system.status.OSHISystemInfoCollector.fs [class=oshi.software.os.linux.LinuxFileSystem] <----- field that is causing the problem


```

File: src/main/src/main/java/org/geoserver/system/status/MetricInfo.java [hunk 1]
```diff
@@ -18,7 +18,7 @@
     LOGICAL_CPUS("CPU", 101, "Number of logical CPUs"),
     RUNNING_PROCESS("CPU", 102, "Number of running process"),
     RUNNING_THREADS("CPU", 103, "Number of running threads"),
-    CPU_LOAD("CPU", 104, "CPU average load", "%"),
+    // removed from OSHI    CPU_LOAD("CPU", 104, "CPU average load", "%"),
```
Comment (mprins) [inline src/main/src/main/java/org/geoserver/system/status/MetricInfo.java:21]: should this just be removed from the enum?

Comment (aaime) [reply src/main/src/main/java/org/geoserver/system/status/MetricInfo.java:21]: Yep please

**Comment by dbwiddis** [comment]
> BTW, werent' there licensing changes in OSHI making it compatible with GPL?

Yes, that was precisely one of the reasons for the 2019 change from EPL to MIT.  Interestingly I was looking at that exact license exemption yesterday and wondering why it was still there.

File: src/main/src/main/java/org/geoserver/system/status/MetricInfo.java [hunk 2]
```diff
@@ -18,7 +18,7 @@
     LOGICAL_CPUS("CPU", 101, "Number of logical CPUs"),
     RUNNING_PROCESS("CPU", 102, "Number of running process"),
     RUNNING_THREADS("CPU", 103, "Number of running threads"),
-    CPU_LOAD("CPU", 104, "CPU average load", "%"),
+    // 104 removed from OSHI
```
Comment (dbwiddis) [inline src/main/src/main/java/org/geoserver/system/status/MetricInfo.java:21]: What was removed was OSHI tracking its own ticks here. You could keep track of the ticks yourself imilar to how you do for `PER_CPU_LOAD` and keep this.

File: src/main/src/main/java/org/geoserver/system/status/OSHISystemInfoCollector.java [hunk 3]
```diff
@@ -184,22 +187,11 @@ public OSHISystemInfoCollector() {
                         si = Collections.singletonList(mv);
                         break;
                     }
-                case CPU_LOAD:
```
Comment (dbwiddis) [inline src/main/src/main/java/org/geoserver/system/status/OSHISystemInfoCollector.java:187]: Rather than removing this entirely you could use an `oldTicks` value (like you do below on line 194 for per-CPU) and pass to the `getSystemCpuLoadBetweenTicks()` method.  Alternately you could duplicate the per-cpu logic and sum up the arrays to get a system average.

**Comment by mprins** [comment]
thanks @dbwiddis  I will update our license file as well

File: src/main/src/main/java/org/geoserver/system/status/OSHISystemInfoCollector.java (hunk 3 above)
Comment (mprins) [reply src/main/src/main/java/org/geoserver/system/status/OSHISystemInfoCollector.java:187]: :+1:

**Comment by mprins** [comment]
MacOS failure seems unrelated:
```
[INFO] Results:
[INFO] 
Error:  Failures: 
Error:    BufferedImageLegendGraphicOutputFormatTest.testMultipleLayers:242->GeoServerSystemTestSupport.assertPixel:1774 expected:<java.awt.Color[r=0,g=0,b=0]> but was:<java.awt.Color[r=12,g=12,b=12]>
[INFO] 
Error:  Tests run: 1270, Failures: 1, Errors: 0, Skipped: 3
```
https://github.com/geoserver/geoserver/pull/4547/checks?check_run_id=1310738388#step:5:15747

**Comment by px6652586** [comment]
Does anyone know that the incomplete data obtained by calling seed for map slicing? A lot of slicing is missing. Is it a bug or something else?
